doc/extdev/markupapi.rst

```diff
@@ -85,68 +87,83 @@ using :meth:`.Sphinx.add_directive` or :meth:`.Sphinx.add_directive_to_domain`.
       The state and state machine which controls the parsing.  Used for
       ``nested_parse``.
 
+.. seealso::
 
-ViewLists
-^^^^^^^^^
+   `Creating directives`_ HOWTO of the Docutils documentation
 
-Docutils represents document source lines in a class
-``docutils.statemachine.ViewList``.  This is a list with extended functionality
--- for one, slicing creates views of the original list, and also the list
-contains information about the source line numbers.
+   .. _Creating directives: https://docutils.sourceforge.io/docs/howto/rst-directives.html
 
-The :attr:`Directive.content` attribute is a ViewList.  If you generate content
-to be parsed as ReST, you have to create a ViewList yourself.  Important for
-content generation are the following points:
 
-* The constructor takes a list of strings (lines) and a source (document) name.
+.. _parsing-directive-content-as-rest:
 
-* The ``.append()`` method takes a line and a source name as well.
+Parsing directive content as reStructuredText
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Many directives will contain more markup that must be parsed.
+To do this, use one of the following APIs from the :meth:`Directive.run` method:
 
+* :py:meth:`.SphinxDirective.parse_content_to_nodes()`
+* :py:meth:`.SphinxDirective.parse_text_to_nodes()`
+* :py:func:`sphinx.util.parsing.nested_parse_to_nodes()`
 
-Parsing directive content as ReST
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The first method parses all the directive's content as markup,
+whilst the second only parses the given *text* string.
+Both methods return the parsed Docutils nodes in a list.
 
-Many directives will contain more markup that must be parsed.  To do this, use
-one of the following APIs from the :meth:`Directive.run` method:
+The methods are used as follows:
 
-* ``self.state.nested_parse``
-* :func:`sphinx.util.nodes.nested_parse_with_titles` -- this allows titles in
-  the parsed content.
+.. code-block:: python
 
-Both APIs parse the content into a given node. They are used like this::
+   def run(self) -> list[Node]:
+       parsed = self.parse_content_to_nodes()
+       # or
+       parsed = self.parse_text_to_nodes('spam spam spam')
+       # or
+       parsed = nested_parse_to_nodes(self.state, 'spam spam spam')
```

Comment:
It is not clear here why you would use this instead of `parse_text_to_nodes`?

I imagine you should generally always use `parse_text_to_nodes`, unless you wanted to use `keep_title_context`?

Reply:
I've removed the reference to ``parse_text_to_nodes``.